### PR TITLE
Change getVariant behavior to conform to 0.6

### PR DESCRIPTION
**Pull request by dcolligan**
Issue #532

### CODE REVIEW

File: tests/datadriven/test_variants.py [hunk 1]
```diff
@@ -187,15 +186,8 @@ def _verifyVariantsCallSetIds(self, searchVariants, searchsampleIds):
     def testSearchAllVariants(self):
         self._verifyVariantsCallSetIds(None, self.vcfSamples[:1])
 
-    def testSearchCallSetIdsSystematic(self):
```
Comment (jeromekelleher) [inline tests/datadriven/test_variants.py]: Why remove this whole test case? I think the commend just refers to the ``len(callSetIds) == 0`` bit. The test case should still be good.

File: tests/datadriven/test_variants.py [hunk 2]
```diff
@@ -349,6 +341,44 @@ def convertPyvcfNumber(number):
...
+        end = datamodel.PysamDatamodelMixin.vcfMax
+        callSetIds = variantSet.getCallSetIds()
+        someCallSetIds = callSetIds[0:3]
+        for referenceName in self._referenceNames:
+            # passing None as the callSetIds argument should be equivalent
+            # to passing all of the possible callSetIds as an argument
+            noneRecords = variantSet.getVariants(
+                referenceName, start, end, None, None)
+            allRecords = variantSet.getVariants(
+                referenceName, start, end, None, callSetIds)
```
Comment (jeromekelleher) [inline tests/datadriven/test_variants.py]: need to check that ``len(noneRecords) == len(allRecords)`` here (zip will run to the shortest of the two iterators).

File: tests/datadriven/test_variants.py [hunk 3]
```diff
@@ -349,6 +341,44 @@ def convertPyvcfNumber(number):
...
+            for record in emptyRecords:
+                self.assertEqual(len(record.calls), 0)
+
+            # passing some callSetIds as the callSetIds argument should
+            # return only those calls
+            someRecords = variantSet.getVariants(
+                referenceName, start, end, None, someCallSetIds)
+            for record in someRecords:
+                for call in record.calls:
+                    compoundId = datamodel.CallSetCompoundId(
```
Comment (jeromekelleher) [inline tests/datadriven/test_variants.py]: I think we can zip(record.calls, someCallSetIds) and assert the ID of the call is equal to the specified ID. The protocol requires that the calls be returned in the same order that they are specified in, so this is safe.

**Comment by jeromekelleher** [comment]
Looks good @dcolligan, thanks. +1 minor modulo the minor comments above.

**Comment by dcolligan** [comment]
@jeromekelleher pushed an update

File: tests/datadriven/test_variants.py [hunk 4]
```diff
@@ -349,6 +346,45 @@ def convertPyvcfNumber(number):
...
+                referenceName, start, end, None, [])
+            for record in emptyRecords:
+                self.assertEqual(len(record.calls), 0)
+
+            # passing some callSetIds as the callSetIds argument should
+            # return only those calls
+            someRecords = variantSet.getVariants(
+                referenceName, start, end, None, someCallSetIds)
+            for record in someRecords:
+                for call, someId in zip(record.calls, someCallSetIds):
```
Comment (jeromekelleher) [inline tests/datadriven/test_variants.py]: Still need to check the ``len``s of these lists: they could be unequal.

File: tests/datadriven/test_variants.py [hunk 5]
```diff
@@ -349,6 +346,45 @@ def convertPyvcfNumber(number):
...
+                    compoundId = datamodel.CallSetCompoundId(
+                        variantSetCompoundId, call.callSetName)
```
Comment (jeromekelleher) [inline tests/datadriven/test_variants.py]: Why bother with the compound ID here? Why not just
```
self.assertEqual(call.id, someId)
```

**Comment by dcolligan** [comment]
@jeromekelleher pushed an update

**Comment by jeromekelleher** [comment]
Thanks @dcolligan, merging.

**Pull request merged**